From 4b361ee7c2a7d91d9bbab3b69ec3e702fc391ebb Mon Sep 17 00:00:00 2001
From: Liu Xiaowen <b37945@freescale.com>
Date: Tue, 24 Dec 2013 19:11:33 +0800
Subject: [PATCH 485/502] ENGR00215174-2 fix wifi display issue for fsl hwc.

fix wifi display issue for fsl hwc.
refine some code.

Signed-off-by: Liu Xiaowen <b37945@freescale.com>
---
 mx6/hwcomposer/hwcomposer.cpp | 26 +++++++++++---------------
 1 file changed, 11 insertions(+), 15 deletions(-)

diff --git a/mx6/hwcomposer/hwcomposer.cpp b/mx6/hwcomposer/hwcomposer.cpp
index 39e5782..ff625ce 100644
--- a/mx6/hwcomposer/hwcomposer.cpp
+++ b/mx6/hwcomposer/hwcomposer.cpp
@@ -133,6 +133,14 @@ static bool checkG2dProcs(struct hwc_context_t* ctx,
         return false;
     }
 
+    hwc_layer_1_t* targetLayer = &list->hwLayers[list->numHwLayers-1];
+    struct private_handle_t *targetHandle;
+    targetHandle = (struct private_handle_t *)targetLayer->handle;
+    if (targetHandle == NULL) {
+        ALOGI("prepare: targetHandle is null");
+        return false;
+    }
+
     hwc_layer_1_t* layer = NULL;
     for (size_t i=0; i<list->numHwLayers-1; i++) {
         layer = &list->hwLayers[i];
@@ -164,14 +172,6 @@ static int hwc_prepare_physical(struct hwc_context_t* ctx, int disp,
         return 0;
     }
 
-    hwc_layer_1_t* targetLayer = &list->hwLayers[list->numHwLayers-1];
-    struct private_handle_t *targetHandle;
-    targetHandle = (struct private_handle_t *)targetLayer->handle;
-    if (targetHandle == NULL) {
-        ALOGI("prepare: targetHandle is null");
-        return 0;
-    }
-
     if (!ctx->mDispInfo[disp].connected) {
         ALOGE("physical display:%d is diconnected", disp);
         ctx->mDispInfo[disp].mG2dProcs = false;
@@ -233,10 +233,6 @@ static int hwc_set_physical(struct hwc_context_t* ctx, int disp,
     hwc_layer_1_t* targetLayer = &list->hwLayers[list->numHwLayers-1];
     struct private_handle_t *targetHandle;
     targetHandle = (struct private_handle_t *)targetLayer->handle;
-    if (targetHandle == NULL) {
-        ALOGI("hwc_set: targetHandle is null");
-        return 0;
-    }
 
     if (!ctx->mDispInfo[disp].mG2dProcs) {
         if (targetHandle != NULL && ctx->mDispInfo[disp].connected) {
@@ -274,7 +270,7 @@ static int hwc_set_physical(struct hwc_context_t* ctx, int disp,
         int fenceFd = layer->acquireFenceFd;
         if (fenceFd > 0) {
             ALOGI("fenceFd:%d", fenceFd);
-            //sync_wait(fenceFd, -1);
+            sync_wait(fenceFd, -1);
             close(fenceFd);
             layer->acquireFenceFd = -1;
         }
@@ -336,13 +332,13 @@ static int hwc_set_virtual(struct hwc_context_t* ctx, int disp,
         close(fenceFd);
         list->outbufAcquireFenceFd = -1;
     }
-    list->retireFenceFd = -1;
+
     for (size_t i=0; i<list->numHwLayers-1; i++) {
         layer = &list->hwLayers[i];
         int fenceFd = layer->acquireFenceFd;
         if (fenceFd > 0) {
             ALOGI("fenceFd:%d", fenceFd);
-            //sync_wait(fenceFd, -1);
+            sync_wait(fenceFd, -1);
             close(fenceFd);
             layer->acquireFenceFd = -1;
         }
-- 
1.8.0

